containers/ftp/s3_files.rb

```diff
@@ -0,0 +1,103 @@
+require 'aws-sdk-s3'
+require 'digest/sha2'
+require 'tempfile'
+require 'fileutils'
+require 'logger'
+load './utils.rb'
+
+class S3Files
```

Comment:
This is a more paranoid s3 download class bases on all the various errors that have been handled in fixer.

Reply:
added to code comments